framework/src/interfaces/Coupleable.C

```diff
@@ -450,8 +450,75 @@ Coupleable::coupledValue(const std::string & var_name, unsigned int comp) const
   }
 }
 
+template <typename T>
+const typename OutputTools<T>::VariableValue &
+Coupleable::vectorTagValueHelper(const std::string & var_name,
+                                 const TagID tag,
+                                 const unsigned int comp) const
+{
+  const auto * const var = getVarHelper<MooseVariableField<T>>(var_name, comp);
+  if (!var)
+    mooseError(var_name, ": invalid variable name for coupledVectorTagValue");
+  checkFuncType(var_name, VarType::Ignore, FuncAge::Curr);
+
+  if (!_c_fe_problem.vectorTagExists(tag))
+    mooseError("Attempting to couple to vector tag with ID ",
+               tag,
+               "in ",
+               _c_name,
+               ", but a vector tag with that ID does not exist");
+
+  const_cast<Coupleable *>(this)->addFEVariableCoupleableVectorTag(tag);
+
+  if (_c_nodal)
+    return var->nodalVectorTagValue(tag);
+  else
+    return var->vectorTagValue(tag);
+}
+
+template <typename T>
+void
+Coupleable::requestStates(const std::string & var_name,
+                          const TagName & tag_name,
+                          const unsigned int comp)
+{
+  auto var =
+      const_cast<MooseVariableField<T> *>(getVarHelper<MooseVariableField<T>>(var_name, comp));
+  if (!var)
+    mooseError(var_name, ": invalid variable name for tag coupling");
+
+  auto & var_sys = var->sys();
+  if (tag_name == Moose::OLD_SOLUTION_TAG)
+    var_sys.needSolutionState(1);
+  else if (tag_name == Moose::OLDER_SOLUTION_TAG)
+    var_sys.needSolutionState(2);
+}
```

Comment:
You mean a `MooseEnum`? Because we are comparing against `TagNames`, e.g. `std::string`. These are not directly convertible to integers unless maybe we used a `MooseEnum`

Reply:
Or are you suggesting that we should change the `needSolutionState` signature to take an `enum` instead of an integer?

Reply:
either way we would still need a conversion from string to integral type